host/tests/unit/search-index-test.ts

```diff
@@ -507,6 +508,39 @@ module('Unit | search-index', function () {
     );
   });
 
+  test("indexing identifies an instance's card references", async function (assert) {
+    let realm = TestRealm.create({
+      'person-1.json': {
+        data: {
+          type: 'card',
+          attributes: {
+            firstName: 'Mango',
+          },
+          meta: {
+            adoptsFrom: {
+              module: `${testModuleRealm}person`,
+              name: 'Person',
+            },
+          },
+        },
+      },
+    });
+    let indexer = realm.searchIndex;
+    await indexer.run();
+    let refs = (await indexer.searchEntry(new URL(`${testRealmURL}person-1`)))
+      ?.refs;
+    assert.deepEqual(
+      [...refs!.keys()].sort(),
+      [
+        `${testModuleRealm}person/Person`,
+        `${baseRealm.url}card-api/Card`,
+        `${baseRealm.url}integer/default`,
+        `${baseRealm.url}string/default`,
+      ],
+      'the card references for the instance are correct'
+    );
```

Comment:
I made the reference tracking be complete and will include references in external realms.